### DISCUSSION TITLE
说起来，arkweb应该也是兼容jsbridge的吧，扫一扫等问题应该可以解决

### DISCUSSION THREAD

**Discussion by IShiraiKurokoI** [tianlongbaobao/idut-HarmonyOS-Next#1] [Ideas]
看到了一个这个文章
https://developer.huawei.com/consumer/cn/doc/best-practices-V5/bpta-arkweb_rendering_framework-V5
![image](https://github.com/user-attachments/assets/8c432a5b-1204-4d36-97e7-862784dabae9)
i大工原生的扫码和本地交互的操作是通过jsbridge+本地微哨SDK预定义command实现的，总体上就是h5告诉本地我要干什么，本地去执行，将数据再返回webview,可以参考我的
相关指令列表：
https://github.com/IShiraiKurokoI/DLUTToolBoxMobileV2/tree/main/app/src/main/java/com/Shirai_Kuroko/DLUTMobile/UI/InnerBrowsers/SDK
指令调用分配与结果返回：
https://github.com/IShiraiKurokoI/DLUTToolBoxMobileV2/blob/main/app/src/main/java/com/Shirai_Kuroko/DLUTMobile/UI/InnerBrowsers/SDK/BrowserProxy.java
向内置浏览器添加javascript proxy（指令名和proxy名是不能变的，这个是h5里写的调用点）
https://github.com/IShiraiKurokoI/DLUTToolBoxMobileV2/blob/main/app/src/main/java/com/Shirai_Kuroko/DLUTMobile/UI/InnerBrowsers/BrowserActivity.java#L105

我推测如果你自己实现对应的指令，应该扫码啊，版本过低啊之类的都能解决了，另外的直接调webview api的可能要用别的方法实现，不过我听说这个arkweb也是chrome内核，理论上或许有解决办法？

**Comment by tianlongbaobao** [comment]
感谢大佬的idea，之前看其它大厂比如美团他们的扫码是直接接入了鸿蒙的Scan Kit这个系统级Kit，我在尝试后发现扫码失败，看开发文档后推测i大工这边并没有接入这个Kit，在开发玉兰卡板块的时候使用webview扫码报错："Uncaught ReferenceError: __nativeWhistleProxy is not defined", source: https://card.m.dlut.edu.cn/js/ws/whistle-6.8.4.js?202202180931 (136)，看来是没有调用javascript proxy这个问题。版本过低这个问题目前只会出现在照片采集的拍摄，估计整个相机部分的功能都要做适配，但是在照片采集的相册却意外都拉起了相册和相机，这让我对服务器后端的适配有所怀疑。之后这一点我会补上！

**Reply by IShiraiKurokoI** [reply]
@tianlongbaobao 
版本过低那边确实，我想了下，可能是因为webview没有获取到权限（当时我好像也遇到过这个问题），他是直接通过webview的接口去访问摄像头的，可能是webview默认没有给权限
Android上的实现是
https://github.com/IShiraiKurokoI/DLUTToolBoxMobileV2/commit/6d74ac208d5ce1055d1047e76b4f888ac9dad3b2#diff-3c5fd3f8b87b8505cdb5fe4e6a3fad59c7962ef9e82da9d96271679836cc50c0R341

对应在arkweb里的实现是
https://developer.huawei.com/consumer/cn/doc/harmonyos-guides/web-rtc-V5
![image](https://github.com/user-attachments/assets/c2f585fd-5e0a-4fb4-8912-e568e8c7741c)


应该就能解决了

相册那边，因为调用的是默认系统相册接口，默认webview权限是允许访问相册的，所以是调起的系统filechooser/相册程序，在系统程序里是系统程序可以自己调用相机拍照，不是i大工做的适配

**Comment by tianlongbaobao** [comment]
目前这个问题在2.4.0版本已解决，再次感谢大佬！
